### ISSUE TITLE
How to handle conversion from unversioned to versioned type?

### ISSUE THREAD

**Issue by lentzi90** [kubernetes/code-generator#142]
I'm trying to convert a `*[]string` to `*[]SecurityGroupParam` in [this PR](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1291). This seemed quite straight forward when going from one version to the next, but with multiple versions I seem to have found a corner case where I'm stuck.

There are 3 relevant versions: v1alpha4, v1alpha5 and v1alpha6. In v1alpha4 and v1alpha5 the relevant field is `*[]string`. In v1alpha6 we are trying to change this to the versioned `*[]SecurityGroupParam` type instead.

For going from v1alpha5 to v1alpha6 I would create a function named `Convert_Slice_string_To_Slice_v1alpha6_SecurityGroupParam`. The issue is that the function name would be the same also for v1alpha4 to v1alpha6. If I define it the same function in both versions the generator complains that there is a duplicate.

So then I tried defining the function in only v1alpha5. Then the generator is happy, but the compiler complains because it is importing the v1alpha4 package in v1alpha5.

How can I work around this?

In a nutshell:
```
v1alpha4:
SecurityGroups *[]string
Convert_Slice_string_To_Slice_v1alpha6_SecurityGroupParam

v1alpha5:
SecurityGroups *[]string
Convert_Slice_string_To_Slice_v1alpha6_SecurityGroupParam

v1alpha6
SecurityGroups *[]SecurityGroupParam
```

**Comment by lentzi90** [comment]
I think I figured it out. By adding the conversion function in v1alpha6 I avoided all the issues. No need to import anything extra and no need to duplicate.
